lib/puppet/util/windows/process.rb

```diff
@@ -9,6 +9,9 @@ module Puppet::Util::Windows::Process
   WAIT_TIMEOUT = 0x102
 
   def execute(command, arguments, stdin, stdout, stderr)
+    cwd = arguments[:cwd]
+    Dir.chdir(cwd) if cwd
```

Comment:
Use block format because on Windows, we will not be executing this stuff in a child process so the `cd` will move Puppet to the `cwd` instead of just the child.